src/devices/Sht4x/Sht4x.cs

```diff
@@ -0,0 +1,154 @@
+    /// </summary>
+    [Interface("Humidity and Temperature Sensor SHT4x")]
+    public sealed class Sht4x : IDisposable
+    {
+        /// <summary>
+        /// The default I²C address of this device.
+        /// </summary>
+        public const int DefaultI2cAddress = 0x44;
+
+        private readonly I2cDevice _device;
+        private RelativeHumidity _lastHum;
+        private Temperature _lastTemp;
+
+        /// <summary>
+        /// The level of repeatability to use when measuring relative humidity.
+        /// </summary>
+        [Property]
+        public Sht4xRepeatability Repeatability { get; set; } = Sht4xRepeatability.High;
+
+        /// <summary>
+        /// The most recent relative humidity measurement.
+        /// </summary>
+        [Telemetry]
+        public RelativeHumidity Humidity
+        {
+            get
+            {
+                ReadHumidityAndTemperature(Repeatability);
+                return _lastHum;
+            }
+        }
+
+        /// <summary>
+        /// The most recent temperature measurement.
+        /// </summary>
+        [Telemetry]
+        public Temperature Temperature
+        {
+            get
+            {
+                ReadHumidityAndTemperature(Repeatability);
+                return _lastTemp;
+            }
+        }
+
+        /// <summary>
+        /// Instantiates a new <see cref="Sht4x"/>.
+        /// </summary>
+        /// <param name="device">The I²C device to operate on.</param>
+        public Sht4x(I2cDevice device)
+        {
+            _device = device;
+            Reset();
+        }
+
+        /// <inheritdoc/>
+        public void Dispose() =>
+            _device.Dispose();
+
+        /// <summary>
+        /// Resets the device.
+        /// </summary>
+        public void Reset()
+        {
+            _device.WriteByte(0x94);
+            Thread.Sleep(1);
+        }
+
+        /// <summary>
+        /// Reads relative humidity and temperature.
+        /// </summary>
+        /// <returns>
+        /// A tuple of relative humidity and temperature.
+        /// If a CRC check failed for a measurement, it will be <see langword="null"/>.
+        /// </returns>
+        public (RelativeHumidity? RelativeHumidity, Temperature? Temperature) ReadHumidityAndTemperature(Sht4xRepeatability repeatability = Sht4xRepeatability.High)
+        {
+            TimeSpan delay = BeginReadHumidityAndTemperature(repeatability);
+            Thread.Sleep(delay);
+            return EndReadHumidityAndTemperature();
+        }
+
+        /// <summary>
+        /// Begins reading relative humidity and temperature.
+        /// </summary>
+        /// <returns>
+        /// The time that the caller must wait before calling <see cref="EndReadHumidityAndTemperature"/> to retrieve the measurement.
+        /// </returns>
+        public TimeSpan BeginReadHumidityAndTemperature(Sht4xRepeatability repeatability = Sht4xRepeatability.High)
```

Comment:
Similar comments than the other type, perhaps some of these members should be private, in favor of making the public API surface easier to understand and use.